src/bin/cargo/commands/help.rs

```diff
@@ -53,16 +60,40 @@ pub fn exec(gctx: &mut GlobalContext, args: &ArgMatches) -> CliResult {
         None => subcommand.to_string(),
     };
 
-    if !super::builtin_exec(&subcommand).is_some() {
+    let builtins = all_builtin_commands();
+    let Some(cmd_kind) = builtins.get(&subcommand).cloned() else {
```

Comment:
I see `let-else` being useful for early returns but this is putting major business logic inside of it.